### PR TITLE
Fix pills not correctly tipping onto the floor

### LINKED ISSUE #21109: pill bottle pills disappeared
### Labels

[BUG][MAJOR]

### Description

dumped pills on the escape shuttle floor and they disappeared.

### Steps to reproduce

1. have pills in a bottle and use them by hitting C
2. pills leave bottle and are gone forever. 

### Expected Behavior

expected pills to be dumped on floor

### Additional Information & Screenshots

Reported by: thedude6969
Client version: 515.1641
On server: Goonstation 3 Roleplay: Morty
Active test merges: #20884 and #20898
Round log date: 2024-10-18-10-18
Reported on: 2024-10-18 12:02:20
Round ID: 190920
Map: KONDARU

### PR DESCRIPTION
**Pull request by Egregorious**
## About the PR 

Pills from Chemaster pill bottles will correctly fall to the floor when tipped out.

## Why's this needed? 

I made a mistake with #20872 and pills were being sent to the aether instead of the turf when tipped out.

fixes #21109

### CODE REVIEW

File: code/obj/item/chem_pill_bottle.dm [hunk 1]
```diff
@@ -99,10 +99,11 @@
 
 	proc/tip_out(var/mob/user as mob, atom/location, var/skip_messages = FALSE)
 		var/obj/item/reagent_containers/pill/P = src.create_pill()
+		if (!istype(location, /turf)) location = location.loc
 		if (istype(P))
 			var/i = rand(3,8)
 			while(istype(P) && i > 0)
-				P.set_loc(location.loc)
+				P.set_loc(location)
```
Comment (TobleroneSwordfish) [inline code/obj/item/chem_pill_bottle.dm:106]:
```suggestion
				P.set_loc(get_turf(location))
```

File: code/obj/item/chem_pill_bottle.dm [hunk 2]
```diff
@@ -99,10 +99,11 @@
 
 	proc/tip_out(var/mob/user as mob, atom/location, var/skip_messages = FALSE)
 		var/obj/item/reagent_containers/pill/P = src.create_pill()
+		if (!istype(location, /turf)) location = location.loc
```
Comment (TobleroneSwordfish) [inline code/obj/item/chem_pill_bottle.dm:102]:
```suggestion
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
More robust code if you *always* want it to be a turf

**Comment by Egregorious** [comment]
That makes a lot more sense, thanks!

**Pull request merged**